### DISCUSSION TITLE
[FEATURE REQ] Auto remove torrents

### DISCUSSION THREAD

**Discussion by mizz141** [autobrr/autobrr#1651] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm utilizing autobrr to autosnatch a bunch of torrents, reach Minimum seedtime with them, and then remove.
Altough I've only now realized that the seedtime filter doesn't automatically remove the torrent when the limit is reached, rather put them in a paused state, and not outright remove the torrent including files.

**Describe the solution you'd like**
Have a toggle which says "Remove Torrent and Files after seedtime limit is reached" next to the seedtime limit, so that it can be toggled on and off

**Describe alternatives you've considered**
[Autoremove-Torrents](https://github.com/jerrymakesjelly/autoremove-torrents) and [Cross-seed](https://github.com/cross-seed/cross-seed) offer a similar functionality, but only on the category and tag specified, and not individual torrents themselves IIRC, also I'd rather have not *another* failure point in my system.

**Additional context**
A small Mockup of what this could look like![imaga](https://user-images.githubusercontent.com/20839616/207134360-3ce669a4-eda5-4adc-8152-f83016b7c8a2.png)

**Comment by Yankees4life** [comment]
I would like that too. Wouldn't work with the arrs but when racing for FLs/regbrr, it would work well.  But it would need to remove cross-seeds too

**Comment by zze0s** [comment]
Those values are just sent to the client and we do nothing more with it, it's up to the client to handle it :)

However, built in handling for autoremove has been in the plans since the beginning but there are good 3rd party solutions for that already so it has not been a focus.

**Comment by gravelfreeman** [comment]
I'm using the autoremove-torrents script available [here](https://autoremove-torrents.readthedocs.io/en/latest/config.html).

Although it's working great I would love to see this implemented in autobrr.

It could be added as it's own filter under Filters. The following would be the filter tabs/sections:

### 1. General

- Filter name
- Trackers
- Rules

### 2. Filters

- **all_trackers / all_categories / all_status:** Choose all the trackers/categories/status.
- **categories:** Choose torrents in these categories.
- **excluded_categories:** Don’t choose torrents in these categories.
- **trackers:** Choose torrents in these trackers.
- **excluded_trackers:** Don’t choose torrents in these trackers.
- **status:** Choose torrents in these status. Available status is as follows:  Downloading, Uploading, Checking, Queued, Paused, Stopped, Error, StalledUpload, StalledDownload.
- **excluded_status:** Don’t choose these torrents in these status. Available status is shown in the table above.

### 3. Remove Condition

- _ratio / create_time / downloading_time / seeding_time / max_download / max_downloadspeed / min_uploadspeed / max_average_downloadspeed / min_average_uploadspeed / max_size / max_seeder / max_upload / min_leecher / max_connected_seeder / min_connected_leecher / last_activity / max_progress / upload_ratio_
- ability to add as many remove conditions we want
- ability to add ``conditionA`` or ``conditionB``
- ability to add ``conditionA`` and ``conditionB``
- ability for complex conditions like ``conditionA`` or (``conditionB``+``conditionC``)
- ability to set a  variable to conditions ``<`` ``>`` ``=`` example: ``conditionA > 2`` and ``conditionB > 3600``

### 4. External filters

- Same as autobrr in case someone would want to customize his filter to his liking with a script or webhook.

### 5. Actions

- Delete data and torrent from client
- Orphan to another folder
- **Transfer to another client/folder (local or client)**

I want to add some weight on Actions > Transfer.

This would be a golden feature since seedboxes can get filled very quickly and it's not good for long term retention. It would be great to be able to according to the filter to be able to transfer certain content to another server and to inject it in another torrent client.

Example 1: Someone adds a filter with omegabrr list of Popular TV on their BTN stuff to transfer everything to their home server so they could access that content after the initial swarm without having to keep all the other unpopular TV shows.

Example 2: Someone with an nvme seedbox with only 1-2 TB could setup a transfer, all torrents which last_activity reach 1 hour, to another client that's on another server. This would help to keep a good average seed time and help retention on trackers.

**Comment by gravelfreeman** [comment]
Where can we track the progress on this?
